src/components/DetailView/DetailView.svelte

```diff
@@ -59,11 +65,42 @@
       }).then((rows) => addMissing(rows, sensor))
     : [];
 
+  // The currently selected date range, initially defaults to the smallMultipleTimeSpan.
+  // Modified by onDateRangeChange, and fetched via (non-reactive) getDateRange,
+  // to avoid updating the chart immediately, so it will be used as the initial value the
+  // next time createSpec is called.
+  // $: dateRange = dateRange || $smallMultipleTimeSpan;
+
+  function getDateRange() {
+    if ($detailViewTimeSpan.length === 0) {
+      detailViewTimeSpan.set($smallMultipleTimeSpan);
+    }
+    return $detailViewTimeSpan;
+  }
+
+  function onDateRangeChange(event) {
+    // All signal events are sent to all on:signal handlers (??), so check if this is the right handler.
+    if (event.detail.name !== 'dateRange') {
+      return;
+    }
+    const dr = event.detail.value && event.detail.value.date_value;
+    if (dr && Array.isArray(dr) && dr.length > 0) {
+      console.info('onDateRangeChange', dr);
+      if (
+        (!Number.isNaN(dr[0]) && dr[0] !== $detailViewTimeSpan[0].getTime()) ||
+        (!Number.isNaN(dr[1]) && dr[1] !== $detailViewTimeSpan[1].getTime())
+      ) {
+        detailViewTimeSpan.set([new Date(dr[0]), new Date(dr[1])]);
+        console.info('onDateRangeChange new dates', $detailViewTimeSpan);
```

Comment:
same here

Reply:
ack.  will do.  

BTW I don't think the console messages make the problem worse, though that might be possible.

Reply:
All works correctly now.